fibonacci.js

```diff
@@ -0,0 +1,110 @@
+const resultsListurl = "http://localhost:5050/getFibonacciResults";
+const resultsListElement = document.getElementById("results-list");
+
+function getFib() {
+  mySpinner.classList.remove("visually-hidden");
+  answer.classList.add("visually-hidden");
+  if (getNumber.value > 50) {
+    alertBox.classList.remove("visually-hidden");
+    spinner.classList.add("visually-hidden");
+  }
+  if (getNumber.value < 50) {
+    fetch(`${fibURL}${getNumber.value}`)
+      .then(function (response) {
+        if (!response.ok) {
+          response.text().then((error) => {
+            answer.classList.remove(
+              "visually-hidden",
+              "text-decoration-underline",
+              "fw-bold"
+            );
+            answer.classList.add("text-danger");
+            alertBox.classList.add("visually-hidden");
+            getNumber.classList.add("border-danger");
+            answer.innerText = error;
+            mySpinner.classList.toggle("visually-hidden");
+          });
+        } else return response.json();
+      })
+
+      .then((data) => {
+        answer.innerHTML = data.result;
+      })
+
+      .then((spinner) => {
+        mySpinner.classList.toggle("visually-hidden");
+        answer.classList.remove("visually-hidden", "text-danger");
+        answer.classList.add("text-decoration-underline", "fw-bold");
+        getNumber.classList.remove("border-danger");
+        alertBox.classList.add("visually-hidden");
+      });
+  }
+}
+
+function getResults() {
+  fetch(resultsListurl)
+    .then((response) => response.json())
+    .then((data) => {
+      printResultstoList(data);
+    })
+    .then(fetch(resultsListurl));
+}
+
+function printResultstoList(data) {
+  let myData = data.results;
+  myData.sort((a, b) => b.createdDate - a.createdDate);
+  resultsListElement.innerHTML = "";
+  for (i = 0; i < 10; i++) {
+    resultsListElement.innerHTML += `<li class="pb-2 pt-2 border-bottom border-secondary">The fibonacci of <b>${
+      myData[i].number
+    }</b> is <b>${myData[i].result}</b>. Calculated at: ${new Date(
+      myData[i].createdDate
+    )}</li>`;
+  }
+}
+
+function fibonacci() {
+  if (getNumber.value > 50) {
+    alertBox.classList.remove("visually-hidden");
+    spinner.classList.add("visually-hidden");
+  } else {
+    let x = 1,
+      y = 0,
+      c;
+
+    for (i = 0; i < getNumber.value; i++) {
+      c = x;
+      x = x + y;
+      y = c;
+    }
+    answer.innerHTML = y;
+    answer.classList.remove("visually-hidden", "text-danger");
+    answer.classList.add("text-decoration-underline", "fw-bold");
+    getNumber.classList.remove("border-danger");
+    alertBox.classList.add("visually-hidden");
+  }
+}
+
+button.addEventListener("click", fibonacci);
+button.removeEventListener("click", getFib);
+button.removeEventListener("click", getResults);
+
+let checkBox = document.querySelector("input[name=checkbox");
+checkBox.addEventListener("change", (e) => {
```

Comment:
Although your code works, a better practice would be to add an event listener only for 'click' event on the "is" button and then determine if the checkbox value is true or false, instead of adding and removing 3 event listeners each time.